src/enclave/enclave.h

```diff
@@ -300,7 +300,7 @@ namespace ccf
                 time_now - last_tick_time);
             if (elapsed_ms.count() > 0)
             {
-              last_tick_time = time_now;
+              last_tick_time += elapsed_ms;
```

Comment:
Tiny change, big effect. Accumulating errors!

`last_tick_time` and `time_now` are in microseconds. But we want to avoid ticking anything more often than every 1ms, so we wait for the `elapsed_ms` to go `> 0` before ticking. Assigning `time_now` means we're losing any overflow from the rounding done on line 298. If we got a 1.5ms, we tick these systems saying 1ms elapsed. Then a second 1.5ms comes in, and we tick exactly 1ms again. 3ms total real-time gone, only 2ms ticked to all of the listening systems.

The fix is to 'save' that overflow by not assigning it to `last_tick_time` - we only advance it by the `elapsed_ms` we're reporting to the other systems, so it stays in-sync with those counts.

The effect of this was that all of the tasks triggered by the timers below (including `thread_messaging`, which is _everything else_), were ticking slow. So a task that was supposed to happen in 200ms, would only happen after ~240ms of real-time.